types/stellar-sdk/index.d.ts

```diff
@@ -18,13 +18,20 @@ export class Account {
 
 export class CallBuilder<T extends Record> {
     constructor(serverUrl: string)
-    call(): Promise<CollectionPage<T>>;
     cursor(cursor: string): this;
     limit(limit: number): this;
     order(direction: 'asc' | 'desc'): this;
     stream(options?: { onmessage?: (record: T) => void, onerror?: (error: Error) => void }): () => void;
 }
 
+export class CollectionCallBuilder<T extends Record> extends CallBuilder<T> {
+    call(): Promise<CollectionPage<T>>;
+}
+
+export class PagedCallBuilder<T extends Record> extends CallBuilder<T> {
```

Comment:
Is this actually paged if it just contains a single record? Could the AccountsCallBuilder just extend CallBuilder<T> directly? If you left call as part of CallBuilder but changed it to return Promise<T> (sorry on the road now and can’t test this out myself at the moment).